network/src/transport.rs

```diff
@@ -60,20 +61,19 @@ impl From<u32> for ConnectionId {
 
 /// Generator of unique ConnectionIds.
 struct ConnectionIdGenerator {
-    next: ConnectionId,
+    ctr: AtomicU32,
 }
 
 impl ConnectionIdGenerator {
-    fn new() -> ConnectionIdGenerator {
+    const fn new() -> ConnectionIdGenerator {
```

Comment:
how can this be a const fn and why make it a const fn oO?

Reply:
this let's you do

```rust
static CONNECTION_ID_GENERATOR: ConnectionIdGenerator = ConnectionIdGenerator::new();
```

without lazy static. it can be a `const fn` b/c `AtomicU32::new` is a `const fn`

Reply:
interesting!